documents/final-submission-document/chapters/3_concept.tex

```diff
@@ -10,6 +10,88 @@ \section{Architecture}
+\label{fig:architecturecontainerrichclient}
+\end{figure}
+
+A web application has the significant advantage to be easily installable, upgradable and has minimal requirements on the user's side (i.e. only requires a web browser to be executed).
+Therefore, we decided to implement the XMPP grid broker as web application.
+
+\subsection{Web Application Topology}
+
+To manage the Controller from our interface, we considered the implementation of either directly connecting to the XMPP server over WebSockets~\cite{rfc7395} or HTTP (BOSH~\cite{xep-0124}), or to communicate indirectly with the XMPP server via custom WebAPI Proxy.
+These topologies are illustrated in Figure~\ref{fig:architecturecontainerwebapplication} and Figure~\ref{fig:architecturecontainerwebproxy}.
+
+As elaborated in the according design decision (see Appendix~\ref{sec:architectural-decisions}), we decided for the direct connection via WebSockets.
+This topology simplifies the implementation and deployment of the application in comparison to a WebAPI Proxy.
+Additionally, WebSockets offer stateful TCP-sockets to exchange data with the XMPP server in contrast to BOSH, which uses HTTP long polling to emulate a bidirectional stream~\cite{xep-0124}.
+
+Using the XMPP Service Discovery~\cite{xep-0030}, the XMPP server may be queried for supported features, so that only supported functionality is presented to the application user.
+
+\begin{figure}[h]
+\centering
+\includegraphics[width=0.7\linewidth]{resources/architecture_container_webapplication}
+\caption[Architecture Container Diagram: Web Application]{Architecture Container Diagram showing the web application topology with WebSockets or BOSH.}
+\label{fig:architecturecontainerwebapplication}
+\end{figure}
+
+\begin{figure}[h]
+\centering
+\includegraphics[width=0.7\linewidth]{resources/architecture_container_proxy.pdf}
+\caption[Architecture Container Diagram: Web Proxy]{Architecture Container Diagram showing the WebAPI Proxy topology.}
+\label{fig:architecturecontainerwebproxy}
+\end{figure}
+
+
+\subsection{Authentication and Connection Security}
+
+XMPP uses SASL as authentication mechanism~\cite{rfc6120}.
+To authenticate against the XMPP Grid Controller, we decided to use the SASL EXTERNAL~\cite{rfc4422} mechanism whenever possible to authenticate the client.
+
+We decided against the alternative SASL authentication method, SASL SCRAM~\cite{rfc7677}, that is also recommended by in the XMPP grid draft~\cite{ietf-mile-xmpp-grid-05}.
+As described in the corresponding architectural decision (Appendix~\ref{sec:architectural-decisions}), the main reason for SASL EXTERNAL is its higher level of security and and its relatively simple scaling capabilities.
+
+SASL EXTERNAL implies that the authentication takes place on a higher layer than the actual XMPP protocol. In our case, this implies authentication over TLS, i.e.~X.509 end-user certificates as specified in RFC6120~\cite{rfc6120}.
```

Comment:
This section is a bit confusing...TLS is (in my perception) the layer below XMPP. Did I misunderstand this?